### PR TITLE
New transformer for node-full_width_banner_alert

### PR DESCRIPTION
**Pull request by jhonny218**
## Description
Created a new transformer for `node-full_width_banner_alert`

## Testing done
Compared to graphQL output after running `node script/cms/test-bundle-transformers.js --bundle "node-full_width_banner_alert" --print 1`

## Acceptance criteria
- [ ]

## Definition of done
- [x] A link has been provided to the originating GitHub issue (or connected to it via ZenHub)
- [x] No sensitive information (i.e. PII/credentials/internal URLs/etc.) is captured in logging, hardcoded, or specs

### CODE REVIEW

File: src/site/stages/build/process-cms-exports/schemas/input/node-full_width_banner_alert.js [hunk 1]
```diff
@@ -0,0 +1,62 @@
...
+      type: 'array',
+      maxItems: 1,
+      items: {
+        type: 'object',
+        properties: {
+          alias: { type: 'string' },
+        },
+        required: ['alias'],
+      },
+    },
```
Comment (bkjohnson) [inline src/site/stages/build/process-cms-exports/schemas/input/node-full_width_banner_alert.js:30]: We have a common `path` schema that I think you could use here like this:

```js
path: { $ref: 'RawPath' },
```

File: src/site/stages/build/process-cms-exports/schemas/output/node-full_width_banner_alert.js [hunk 2]
```diff
@@ -1,3 +1,53 @@
...
+      items: {
+        $ref: 'Paragraph',
+      },
+    },
+  },
+  required: [
+    'title',
+    'changed',
+    'entityUrl',
+    // 'path',
```
Comment (bkjohnson) [inline src/site/stages/build/process-cms-exports/schemas/output/node-full_width_banner_alert.js:38]: Is this something we care about for this entity?

File: src/site/stages/build/process-cms-exports/transformers/node-full_width_banner_alert.js [hunk 3]
```diff
@@ -0,0 +1,60 @@
...
+  utcToEpochTime,
+} = require('./helpers');
+
+const transform = entity => ({
+  entityType: 'node',
+  entityBundle: 'full_width_banner_alert',
+  title: getDrupalValue(entity.title),
+  changed: utcToEpochTime(getDrupalValue(entity.changed)),
+  entityMetatags: createMetaTagArray(entity.metatag.value),
+  path: getDrupalValue(entity.path),
```
Comment (bkjohnson) [inline src/site/stages/build/process-cms-exports/transformers/node-full_width_banner_alert.js:13]: I see that `path` is part of the transformer, so shouldn't it be uncommented in the output schema?

File: src/site/stages/build/process-cms-exports/schemas/input/node-full_width_banner_alert.js [hunk 4]
```diff
@@ -0,0 +1,62 @@
+/* eslint-disable camelcase */
+
+module.exports = {
+  type: 'object',
+  properties: {
+    title: { $ref: 'GenericNestedString' },
+    changed: {
```
Comment (theodur) [inline src/site/stages/build/process-cms-exports/schemas/input/node-full_width_banner_alert.js:7]: I think you can just use `{ $ref: 'GenericNestedString' }` here.

File: src/site/stages/build/process-cms-exports/transformers/node-full_width_banner_alert.js (hunk 3 above)
Comment (jhonnyoddball) [reply src/site/stages/build/process-cms-exports/transformers/node-full_width_banner_alert.js:13]: I have seen it in the inputs and transformers but not in any of the outputs yet

**Comment by jhonnyoddball** [comment]
This transformer uses:
```
fieldBannerAlertVamcs: {
      $ref: 'output/node-vamc_operating_status_and_alerts',
    },
```
However, `node-vamc_operating_status_and_alerts` uses:
```
    fieldBannerAlert: {
      type: 'array',
      items: { $ref: 'output/node-full_width_banner_alert' },
    },
```
They are both calling each other which is creating a problem of who finishes first

File: src/site/stages/build/process-cms-exports/transformers/node-full_width_banner_alert.js (hunk 3 above)
Comment (jhonnyoddball) [reply src/site/stages/build/process-cms-exports/transformers/node-full_width_banner_alert.js:13]: Although, I think the path is used in `entityUrl: { $ref: 'EntityUrl' },` in the ouput

Comment (bkjohnson) [reply src/site/stages/build/process-cms-exports/transformers/node-full_width_banner_alert.js:13]: So if `path` isn't in the transformed output, then shouldn't it be removed from the transformer?

File: src/site/stages/build/process-cms-exports/schemas/input/node-full_width_banner_alert.js [hunk 5]
```diff
@@ -0,0 +1,45 @@
+/* eslint-disable camelcase */
+
+module.exports = {
```
Comment (theodur) [inline src/site/stages/build/process-cms-exports/schemas/input/node-full_width_banner_alert.js:3]: I noticed you didn't add the `created` field here. Was that left out intentionally?

Comment (jhonnyoddball) [reply src/site/stages/build/process-cms-exports/schemas/input/node-full_width_banner_alert.js:3]: It was my initial one but once the problem is solved I'll go back and add what's missing.

File: src/site/stages/build/process-cms-exports/schemas/input/node-full_width_banner_alert.js [hunk 6]
```diff
@@ -0,0 +1,51 @@
...
+      maxItems: 1,
+      items: { $ref: 'EntityReference' },
+    },
+    field_alert_dismissable: { $ref: 'GenericNestedBoolean' },
+    field_alert_email_updates_button: { $ref: 'GenericNestedBoolean' },
+    field_alert_find_facilities_cta: { $ref: 'GenericNestedBoolean' },
+    field_alert_inheritance_subpages: { $ref: 'GenericNestedBoolean' },
+    field_alert_operating_status_cta: { $ref: 'GenericNestedBoolean' },
+    field_alert_type: { $ref: 'GenericNestedString' },
+    field_banner_alert_computdvalues: { $ref: 'GenericNestedString' },
```
Comment (cvalarida) [inline src/site/stages/build/process-cms-exports/schemas/input/node-full_width_banner_alert.js:22]: This is, in fact, the correct field name, despite the two typos. :facepalm: 
image

Comment (jhonnyoddball) [reply src/site/stages/build/process-cms-exports/schemas/input/node-full_width_banner_alert.js:22]: I initially thought It was my typo

File: src/site/stages/build/process-cms-exports/index.js [hunk 7]
```diff
@@ -218,7 +218,7 @@ const entityAssemblerFactory = contentDir => {
 
     // Handle circular references
     const a = findCircularReference(entity, ancestors);
-    if (a) return a;
+    if (a) return a.entity;
```
Comment (theodur) [inline src/site/stages/build/process-cms-exports/index.js:221]: Is this change also related the output schema validation?

Comment (theodur) [reply src/site/stages/build/process-cms-exports/index.js:221]: I noticed it wasn't part of https://github.com/department-of-veterans-affairs/vets-website/pull/13700.

Comment (jhonnyoddball) [reply src/site/stages/build/process-cms-exports/index.js:221]: No, there was a problem in which the child was returned wrapped into an entity object and wan't able to transform the data. This removes the entity wrap and returns a regular entity.

**Review by theodur — APPROVED** [review]
🙇
